src/utils/workshift-utils.ts

```diff
@@ -10,6 +12,140 @@ namespace WorkShiftsUtils {
+  };
+
+  export const getFillingHours = (workShiftsData: EmployeeWorkHoursFormRow[], employee: Employee) => {
+    if (!employee?.regularWorkingHours) return 0;
+    const regularWorkingHours = employee?.regularWorkingHours;
+    const paidWorkHours = parseFloat(getSumOfWorkHoursFromWorkPeriod(workShiftsData, WorkType.PaidWork));
+    if (paidWorkHours < regularWorkingHours) {
+      const vacationHours = parseFloat(getAbsenceHours(workShiftsData, AbsenceType.Vacation));
+      const compensatoryLeaveHours = parseFloat(getAbsenceHours(workShiftsData, AbsenceType.CompensatoryLeave));
+      const sickLeaveHours = parseFloat(getAbsenceHours(workShiftsData, AbsenceType.SickLeave));
+      const fillingHours =
+        regularWorkingHours - paidWorkHours - vacationHours - compensatoryLeaveHours - sickLeaveHours;
+      return fillingHours > 0 ? fillingHours.toFixed(2) : 0;
+    }
+    return 0;
+  };
+
+  export const getOverTimeHoursForDriver = (workShiftsData: EmployeeWorkHoursFormRow[], employee: Employee) => {
+    if (!employee?.regularWorkingHours) return { overTimeHalf: 0, overTimeFull: 0 };
+    const vacationHours = parseFloat(getAbsenceHours(workShiftsData, AbsenceType.Vacation));
+    // Check if the employee has more than 40 hours of vacation in work period and adjust the limit for full overtime
+    const overTimeFullLimit = vacationHours > 40 ? 10 : 12;
+    const regularWorkingHours = employee?.regularWorkingHours;
+    // Calculate paid work hours without training hours (training hours does not cumulate overtime)
+    const paidWorkHours =
+      parseFloat(getSumOfWorkHoursFromWorkPeriod(workShiftsData, WorkType.PaidWork)) -
+      parseFloat(getAbsenceHours(workShiftsData, AbsenceType.Training));
+    if (paidWorkHours <= regularWorkingHours) {
+      return { overTimeHalf: 0, overTimeFull: 0 };
+    }
+    //Check if paid work hours are between regular working hours and regular working hours + 12
+    if (paidWorkHours > regularWorkingHours && paidWorkHours <= regularWorkingHours + overTimeFullLimit) {
+      return { overTimeHalf: (paidWorkHours - regularWorkingHours).toFixed(2), overTimeFull: 0 };
+    }
+    //Check if paid work hours are over regular working hours + 12
+    if (paidWorkHours > regularWorkingHours + overTimeFullLimit) {
+      return {
+        overTimeHalf: overTimeFullLimit,
+        overTimeFull: (paidWorkHours - overTimeFullLimit - regularWorkingHours).toFixed(2),
+      };
+    }
+
+    return { overTimeHalf: 0, overTimeFull: 0 };
+  };
+
+  export const getOverTimeHoursForOfficeAndTerminalWorkers = (workShiftsData: EmployeeWorkHoursFormRow[]) => {
+    const overTimeResult = { overTimeHalf: 0, overTimeFull: 0 };
+
+    // Iterate over each work shift to calculate overtime on a daily basis
+    for (const shift of workShiftsData) {
+      const dailyWorkHours =
+        shift.workShiftHours[WorkType.PaidWork]?.actualHours ??
+        shift.workShiftHours[WorkType.PaidWork]?.actualHours ??
+        0;
```

Comment:
Probably a bug, should this be like
```suggestion
      const dailyWorkHours =
        shift.workShiftHours[WorkType.PaidWork]?.actualHours ??
        shift.workShiftHours[WorkType.PaidWork]?.calculatedHours ??
        0;
```